patches/server/0962-Bell-API.patch

```diff
@@ -0,0 +1,61 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Lama06 <andreasprues36@gmail.com>
+Date: Sun, 22 Jan 2023 11:33:25 +0100
+Subject: [PATCH] Bell API
+
+== AT ==
+public net.minecraft.world.level.block.entity.BellBlockEntity resonating
+public net.minecraft.world.level.block.entity.BellBlockEntity resonationTicks
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java b/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java
+index 6c5ed0fc9f58b9ec22194074a3fd69b387e3b270..81ea1fd6435093e8264d86db75ef8e84fe29455d 100644
+--- a/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java
++++ b/src/main/java/org/bukkit/craftbukkit/block/CraftBell.java
+@@ -9,4 +9,47 @@ public class CraftBell extends CraftBlockEntityState<BellBlockEntity> implements
+     public CraftBell(World world, BellBlockEntity tileEntity) {
+         super(world, tileEntity);
+     }
++
++    // Paper start - Bell API
++    @Override
++    public void ring() {
++        requirePlaced();
++        getTileEntity().onHit(getTileEntity().getBlockState().getValue(net.minecraft.world.level.block.BellBlock.FACING));
++    }
++
++    @Override
++    public void ring(org.bukkit.block.BlockFace direction) {
++        if (!direction.isCartesian()) throw new IllegalArgumentException("A bell can only ring in a cartesian direction");
++        if (direction.getModY() != 0) throw new IllegalArgumentException("A bell cannot ring up or down");
++        requirePlaced();
++        getTileEntity().onHit(CraftBlock.blockFaceToNotch(direction));
++    }
++
++    @Override
++    public boolean isRinging() {
++        return getTileEntity().shaking;
++    }
++
++    @Override
++    public int getTicksRinging() {
++        return getTileEntity().ticks;
++    }
++
++    @Override
++    public org.bukkit.block.BlockFace getRingingDirection() {
++        if (!isRinging()) return null;
++
++        return CraftBlock.notchToBlockFace(getTileEntity().clickDirection);
++    }
++
++    @Override
++    public boolean isResonating() {
++        return getTileEntity().resonating;
++    }
++
++    @Override
++    public int getTicksResonating() {
++        return getTileEntity().resonationTicks;
```

Comment:
I have a bit of an issue with this method.
`resonationTicks` is not reset to 0 until the next time the bell is rang.
A non resonating bell will end up with a 40 tick val on this, even if it is no longer resonating.

Reply:
I now check whether the bell is still resonating and return 0 if it is not